File: src/main/java/com/kishlaly/ta/analyze/tasks/blocks/groups/ThreeDisplays_Buy_EFI_3.java
package com.kishlaly.ta.analyze.tasks.blocks.groups;

import com.kishlaly.ta.analyze.tasks.ThreeDisplays;
import com.kishlaly.ta.analyze.tasks.blocks.TaskBlockJava;
import com.kishlaly.ta.analyze.tasks.blocks.commons.ScreenBasicValidation;

import java.util.ArrayList;
import java.util.List;

/**
 * The first screen is not used.
 * The second one checks the three EFI values:
 * + all three last values are negative
 * + the second from the end value is lower than the third from the end (the lower point of the figure U)
 * + the last value is higher than the second from the end (the right point of the figure U)
 * <p>
 * SL sliding on the average Bollinger band or TP at the top of the channel, if the last quote is not very high
 * <p>
 * Note: Can't flip for shorts
 */
//TODO implement
public class ThreeDisplays_Buy_EFI_3 implements BlocksGroupJava {
    @Override
    public List<TaskBlockJava> blocks() {
        ThreeDisplays.Config.STOCH_CUSTOM = 30;

        return new ArrayList<TaskBlockJava>() {{
            add(new ScreenBasicValidation());

        }};

    }

    @Override
    public String comments() {
        return "EFI draws U below zero";
    }
}

File: src/main/kotlin/com/kishlaly/ta/analyze/tasks/groups/threedisplays/ThreeDisplays_Buy_EFI_3.kt
package com.kishlaly.ta.analyze.tasks.groups.threedisplays

import com.kishlaly.ta.analyze.tasks.blocks.TaskBlock
import com.kishlaly.ta.analyze.tasks.blocks.commons.ScreenBasicValidation
import com.kishlaly.ta.analyze.tasks.groups.AbstractBlocksGroup

/**
 * The first screen is not used.
 * The second one checks the three EFI values:
 * + all three last values are negative
 * + the second from the end value is lower than the third from the end (the lower point of the figure U)
 * + the last value is higher than the second from the end (the right point of the figure U)
 * <p>
 * SL sliding on the average Bollinger band or TP at the top of the channel, if the last quote is not very high
 * <p>
 * Note: Can't flip for shorts
 */
//TODO implement
class ThreeDisplays_Buy_EFI_3 : AbstractBlocksGroup() {

    override fun blocks(): List<TaskBlock> {
        return object : ArrayList<TaskBlock>() {
            init {
                add(ScreenBasicValidation())
            }
        }
    }

    override fun comments() = super.comments() + "EFI draws U below zero"

}